uninstall.php

```diff
@@ -75,28 +75,7 @@
 	return;
 }
 
-require_once WEBSTORIES_PLUGIN_DIR_PATH . '/includes/uninstall.php';
-
-if ( is_multisite() ) {
-	\Google\Web_Stories\delete_site_options();
-	$site_ids = get_sites(
-		[
-			'fields'                 => 'ids',
-			'number'                 => '',
-			'update_site_cache'      => false,
-			'update_site_meta_cache' => false,
-		]
-	);
-
-	foreach ( $site_ids as $site_id ) {
-		// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.switch_to_blog_switch_to_blog
-		switch_to_blog( $site_id );
-
-		\Google\Web_Stories\delete_site();
-	}
-
-	restore_current_blog();
-} else {
-	\Google\Web_Stories\delete_site();
-}
-\Google\Web_Stories\delete_stories_user_meta();
+$uninstaller = \Google\Web_Stories\UninstallerFactory::create();
```

Comment:
We no longer need the `\Google\Web_Stories\PluginFactory::create()->register();` call further above, right?

Reply:
No, I think we need that, so the settings class registers.

Reply:
And what is registering the Settings class needed for?

Feels a bit weird to register the plugin here _and_ the uninstaller.

Reply:
@spacedmonkey Would love to get an answer here before resolving.

Still not sure why we still need `\Google\Web_Stories\PluginFactory::create()->register();` in this file and why the `Settings` is needed here.

Reply:
@swissspidy This change was added in by @timarney in https://github.com/GoogleForCreators/web-stories-wp/commit/aa324c02b273e2726a424c8521d1db33bcfe7750. 

It is because of the default value of `$erase = (bool) get_option( \Google\Web_Stories\Settings::SETTING_NAME_DATA_REMOVAL );`

I did not add this change. But either way, it is needed now.

Reply:
I don't think this was because of the default value. The default value is `false` anyway.

IIRC `\Google\Web_Stories\PluginFactory::create()->register();` was added because we needed to register the `Capabilities` service in order to be able to remove caps.

But now the registration is done by the `UninstallerFactory`. It doesn't really make sense to have both the `Plugin` _and_ the `Uninstaller` register the same services twice. For uninstall, only the uninstaller should be registering services.

To put this another way: does the uninstall break if you remove this line? I tend to say _no_.

Reply:
I have removed Uninstaller. A while ago. Please review again.